src/audio_midiout.cpp

```diff
@@ -0,0 +1,387 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <array>
+#include <algorithm>
+#include "audio_midiout.h"
+#include "midisequencer.h"
+#include "output.h"
+#ifdef _WIN32
+#include "platform/windows/midiout_device_win32.h"
+#elif __APPLE__
+#include "platform/macos/midiout_device_coreaudio.h"
+#endif
+
+class GenericMidiOut : public MidiOut, public midisequencer::output {
+public:
+	GenericMidiOut(MidiOutDevice* device);
```

Comment:
This is taking ownership of the device, so change this to take `std::unique_ptr<MidiOutDevice>` by value and remove the calls to release() you have in other places where you call this.